### PR TITLE
Lets you temporarily contain a loose tesla in a pair of insulated gloves

### PR DESCRIPTION
**Pull request by ChesterTheCheesy**
# General Documentation
### Intent of your Pull Request
Lets you wrap a tesla ball inside a pair of insulated gloves, "containing" it inside them for 1 minute, before breaking out.
Also gives an achievement for doing so. The insulated gloves with tesla inside can be coated with another pair of insuls, refreshing the 1 minute timer.

### Why is this change good for the game?
Lets people deal with a tesloose. The insulated gloves with tesla inside can be thrown into deep space, cremated, brought to shuttle during glorious death, etc.

# Wiki Documentation
### Briefly describe your PR and the impacts of it, in layman's terms. 
Lets you "contain" a very dangerous ball of energy inside a pair of insulated gloves, destroying them in process. Tesla will break out after a minute.

### What should players be aware of when it comes to the changes your PR is implementing?
If you feel like a suicidal hero, you can wrap a tesla ball inside a pair of insulated gloves to contain it.

### What general grouping does this PR fall under? 
Engineering tweak

### Are there any aspects of the PR that you would like us not to mention on the Wiki?
This could be hinted on the wiki, but should mostly remain a secret.

# Changelog
:cl:  
rscadd: Allows you to temporarily store a loose tesla ball inside a pair of insuls.
imageadd: adds bloated insulated glove
/:cl:

### CODE REVIEW

**Comment by UselessTheremin** [comment]
lmao how many times you gonna open this pr

**Comment by ChesterTheCheesy** [comment]
Only once because i accidentally deleted my fork :(
But hey, no merge conflicts!

**Comment by patpol4** [comment]
This is funny i like it

**Comment by boba247** [comment]
Seem unbalanced, insulated gloves are not hard to come by. Being able to end the threat of a loose engine with a pair of somewhat uncommon gloves nerfs the tesla escaping. And I know it doesn't completely stop the threat you still have to dispose of the gloves I can't imagine it would be difficult to space a pair of gloves.

**Comment by patpol4** [comment]
> Seem unbalanced, insulated gloves are not hard to come by. Being able to end the threat of a loose engine with a pair of somewhat uncommon gloves nerfs the tesla escaping. And I know it doesn't completely stop the threat you still have to dispose of the gloves I can't imagine it would be difficult to space a pair of gloves.

the tesla ball still dusts you on contact so it's risky, also its funny

**Comment by patpol4** [comment]
What if you lowered the time to like 15 seconds or something? that would probably help with the balance issues of this

**Comment by Ratvar** [comment]
Maybe engine that is piss easy to release should be piss easy to contain?

**Comment by ChesterTheCheesy** [comment]
> 
> 
> What if you lowered the time to like 15 seconds or something? that would probably help with the balance issues of this

While testing, even 30 seconds was not enough to do anything with the tesla. Meanwhile anything above a minute seemed too generous. My aim was making players get rid of the tesla, in a clock-ticking fashion, and a minute seemed the most fair. Although if i see people somehow gathering 5 pairs of insuls to go around the station covered in tesla miniballs (they still follow), i'll happily reduce the time.

**Comment by ChesterTheCheesy** [comment]
> 
> 
> Seem unbalanced, insulated gloves are not hard to come by. Being able to end the threat of a loose engine with a pair of somewhat uncommon gloves nerfs the tesla escaping. And I know it doesn't completely stop the threat you still have to dispose of the gloves I can't imagine it would be difficult to space a pair of gloves.

Getting close to tesla is extremely dangerous, one wrong step and you're either dusted or blown to pieces. Pretty much same applies when you're throwing a bunch of bag of holdings at a loose singularity. Insulated gloves might be easy to get one at the time at roundstart, but as time progresses, and everyone desires their own pair, it's really hard to find more. Keep in mind only REAL insuls work, cheap insuls or combat gloves do not.
Secondly, it's not that easy to dispose of it once you catch it. If the tesla is close to space, you can throw it instantly without much worry. However if the tesla you caught is in the middle of the station, it takes around around a minute to get to loose space, so you really have to hurry. But if you gather extra insuls (presumably hard if the engineering is already blown to pieces, not hard otherwise), it's really up to the player what can be done with it. My idea was making the tesla containing easiest for an engineer (real hero uses his own insuls), but not restricting it to them.
Lastly, throwing tesla into space is a big 50:50 whether it'll loop and get back to station, probably to the non-destroyed part of it.

**Comment by ChesterTheCheesy** [comment]
This is basically as easy as it gets (remember to turn sound on):

https://user-images.githubusercontent.com/71487903/112198469-6ca2bc80-8c0d-11eb-8669-aeefec19a119.mp4

**Comment by boba247** [comment]
Yea seems fair. It'd be cool if it shocks you a few times before breaking out of the gloves.

**Comment by ChesterTheCheesy** [comment]
I might add that later. For now i'll favor the player trying to save the station by himself.

**Comment by Readystorm546** [comment]
> I might add that later. For now i'll favor the player trying to save the station by himself.

Would this not go against the NEW RP Rules?

Avoid pain.

Do not treat your character’s life as if it were a forty-minute round with infinite respawn.
Do not deliberately kill yourself to achieve a goal. Try to keep your character alive and
healthy.

**Comment by ChesterTheCheesy** [comment]
Look man, i'm not forcing people to suddenly go hunt for loose tesla balls with their insuls. I expect as much people do this, as many throw bags of holdings at singulos.
As for the pain and "RP" argument, tesla won't blow you up if there's nothing left to explode, and shocks won't hurt you if you are insulated. At that point the biggest threat is being dusted, which is debatable whether it causes pain or not. Should an atmos tech not rush through through flames to save people? Should miner be prohibited from approaching any lavaland mobs because they cause pain? Of course not. You risk your own life, not rush into danger to die. Dying to tesla won't achieve any goal. That rule rather applies to situation where you can blatantly avoid the risk, or it's obvious your efforts will be for naught. For instance: rushing a nukie with a toolbox or going into space without a suit to save a body and dying in meantime.

I'm not saying that the tesla in insuls is a perfect solution. Nothing is. Many people requested that a loose tesla should have a way to be destroyed like singularity does. In my attempt to make it neither too easy, nor extremely complicated, i've came with this: temporary containment if you succeed in getting close to it. This ensures that tesloosing is still a valid way to hijack, you can't approach it while it's exploding everything. Secondly, catching it inside a glove doesn't immediately fix it. In true way of SS13, the person can do ANYTHING with it, not restricting the player to only one solution. It basically becomes a ticking bomb, and they can either space it, or try to "disarm" it by deleting the glove, be it explosion, recycler, cremation, or else. It can be even be used for evil.
Now some people say the tesla shouldn't be containable at all. I can't make everyone happy, but so far i'm trying my damn best to find middle grounds. So please, if you have any more ideas on how to improve this, or you have an idea for a better way to stop a tesloose, tell me.

**Comment by ChesterTheCheesy** [comment]
Oh and one thing i forgot to mention: I'd prefer admins to send an engineering ERT, give already permadead ghosts a job to fix the tesla, rather than seeing it mysteriously disappear and afterward getting a Centcom message "We've nuked the loose tesla for you, because we are omnipotent.".

**Comment by swissloaf** [comment]
bad gag

**Comment by JamieD1** [comment]
Make a Lightning Rod instead

**Comment by ChesterTheCheesy** [comment]
Why can't we have nice things
